Add patches support to transactions

jeromegn opened the pull request:
This adds previously-unsupported patches tracking to transactions. Patches only worked for `apply_changes` in the past.

jeffa5 commented:
Thanks for this! Would you be able to add some tests to check that this works and continues to work?

I also think we should be able to get around having to query the opset again for each patch by storing the `Prop` of each operation in the `TransactionInner::operations` field and then using that to make the patches.

jeromegn commented:
I added a test(s)!

> I also think we should be able to get around having to query the opset again for each patch by storing the `Prop` of each operation in the `TransactionInner::operations` field and then using that to make the patches.

I've also added this (I think I did it right). I'd like to trigger the test pipeline here to make sure it's all good.

alexjg commented:
>  I'd like to trigger the test pipeline here to make sure it's all good.

Note that you can run all the tests that CI runs locally by running `./scripts/ci/run`

orionz commented:
This is superseded by Op Observer
